src/components/charts/historical.tsx

```diff
@@ -89,113 +110,79 @@ export const HistoricalVolume = React.memo(
           series: [
             {
               type: "line",
-              data: volumeData.map((d) => d.volume),
+              data: data.map((d: any) => props.getCalculatedNumber(d)),
             },
           ],
         });
       }
     };
-
     useEffect(() => {
       if (chartDiv.current) {
         echartsRef.current = echarts.init(chartDiv.current);
       }
-      bonfidaVolumeQuery();
+      bonfidaDataChartQuery();
+      let bonfidaTimer = 0;
+      bonfidaTimer = window.setInterval(
```

Comment:
in case api is slow to respond and the interval is small can you use setTimeout and start it each time after the query is finished.